listenbrainz/webserver/views/settings.py

```diff
@@ -246,8 +247,14 @@ def music_services_connect(service_name: str):
         raise APINotFound("Service %s is invalid." % (service_name,))
 
     data = request.json
+
     if "external_user_id" not in data:
         raise APIBadRequest("Missing 'external_user_id' in request.")
+    
+    
+    lastfm_response = requests.get(f"{current_app.config['LASTFM_API_URL']}?method=user.getinfo&user={data['external_user_id']}&api_key={current_app.config['LASTFM_API_KEY']}&format=json", timeout=5)
+    if lastfm_response.status_code == 404:
+        raise APINotFound("last.fm user not found.")
```

Comment:
Could be useful to add the `data['external_user_id']` in this string ("last.FM user XYZ not found")